### PR TITLE
Add release note for 2.4.0

### PR DESCRIPTION
**Pull request by heemin32**
Signed-off-by: Heemin Kim <heemin@amazon.com>

Add release note for 2.4.0
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-geospatial.release-notes-2.4.0.0.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Version 2.4.0.0 Release Notes
+
+Compatible with OpenSearch 2.4.0
+
+### Features
```
Comment (naveentatikonda) [inline release-notes/opensearch-geospatial.release-notes-2.4.0.0.md:5]: We need to add XYShape in the Features list

Comment (heemin32) [reply release-notes/opensearch-geospatial.release-notes-2.4.0.0.md:5]: Could you point me out the PR. I couldn't find it.

Comment (naveentatikonda) [reply release-notes/opensearch-geospatial.release-notes-2.4.0.0.md:5]: I think there are many PRs as each PR merged to main branch
Use this link to filter the PRs https://github.com/opensearch-project/geospatial/pulls?q=is%3Apr+is%3Aclosed+author%3AVijayanB+shape

Comment (heemin32) [reply release-notes/opensearch-geospatial.release-notes-2.4.0.0.md:5]: @VijayanB Hi Vijay. Could you check the PR list and make sure that they are back ported to 2.x branch and attach v2.4.0 label accordingly?

Comment (heemin32) [reply release-notes/opensearch-geospatial.release-notes-2.4.0.0.md:5]: Updated.

**Review by VijayanB — APPROVED** [review]

**Review by naveentatikonda — APPROVED** [review]

**Pull request merged**